app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt

```diff
@@ -326,3 +339,29 @@ private fun drawMarkers(
             .withData(Gson().toJsonTree(it)))
   }
 }
+
+/**
+ * Initialize the location component of the map.
+ *
+ * @param mapView the MapView to initialize the location component on
+ * @param mapViewModel the MapViewModel to update the camera position
+ */
+private fun initLocationComponent(mapView: MapView, mapViewModel: MapViewModel) {
+  val locationComponentPlugin = mapView.location
+  locationComponentPlugin.updateSettings {
+    this.enabled = true
+    this.locationPuck = createDefault2DPuck(true)
+  }
+  val onIndicatorPositionChangedListener = OnIndicatorPositionChangedListener { location ->
+    mapView.mapboxMap.setCamera(CameraOptions.Builder().center(location).build())
```

Comment:
This is annoying, each time our position Changes the camera is centered around us. 

We're too close to the deadline to change it. So can you already create an issue and mention this in the PR please

Reply:
Actually this could be a feature, we could have two modes one that follow us and one that doesn't. But like this is just too constraining

Reply:
Yes that's what i have planned , add a button to center on the puck and not centering on the puck otherwise. Will be relevant for the user. This is part of the task `handling when user move` for next Sprint.

Reply:
Done !